File: app/src/main/java/com/leadinsource/dailyweightlog/db/Weight.java
package com.leadinsource.dailyweightlog.db;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

import java.util.Date;

/**
 * Created by Matt on 23/02/2018.
 */
@Entity(tableName = DataContract.WeightEntry.TABLE_NAME)
public class Weight {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = DataContract.WeightEntry._ID)
    private int id;

    @ColumnInfo(name = DataContract.WeightEntry.COLUMN_DATE)
    private Date date;

    @NonNull
    @ColumnInfo(name = DataContract.WeightEntry.COLUMN_WEIGHT_IN_KG)
    private float weightInKg;

    @ColumnInfo(name = DataContract.WeightEntry.COLUMN_FAT_PC)
    private float fatPc;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public float getWeightInKg() {
        return weightInKg;
    }

    public void setWeightInKg(float weightInKg) {
        this.weightInKg = weightInKg;
    }

    public float getFatPc() {
        return fatPc;
    }

    public void setFatPc(float fatPc) {
        this.fatPc = fatPc;
    }



}

File: app/src/main/java/com/leadinsource/dailyweightlog/db/Weight.kt
package com.leadinsource.dailyweightlog.db

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.RoomMasterTable.TABLE_NAME

import java.util.Date

/**
 * Created by Matt on 23/02/2018.
 */

const val _ID = "_id"
const val TABLE_NAME = "weights"
const val COLUMN_DATE = "date"
const val COLUMN_WEIGHT_IN_KG = "weight_in_kg"
const val COLUMN_FAT_PC = "fat_pc"

@Entity(tableName = TABLE_NAME)
data class Weight(

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = _ID)
    var id: Int = 0,

    @ColumnInfo(name = COLUMN_DATE)
    var date: Date? = null,

    @ColumnInfo(name = COLUMN_WEIGHT_IN_KG)
    var weightInKg: Float = 0.toFloat(),

    @ColumnInfo(name = COLUMN_FAT_PC)
    var fatPc: Float = 0.toFloat(),
)
